### DISCUSSION TITLE
Support for Retrieving Multiple Users by IDs in supabase-js

### DISCUSSION THREAD

**Discussion by SamuraiT** [supabase/supabase#22173] [Feature Requests] [answered]
#### Description:

As of the current version of `supabase-js`, there is functionality to retrieve a single user by their ID using `supabase.auth.admin.getUserById()`. However, there are common scenarios where developers need to retrieve multiple users by their IDs in a single operation. This functionality is crucial for efficiency and performance, especially when dealing with batch operations or when the frontend needs to display information about multiple users simultaneously.

#### Current Behavior:

To retrieve a user by their ID, we can use:

```javascript
const { data, error } = await supabase.auth.admin.getUserById('userId')
```

However, there's no built-in method to retrieve multiple users by IDs, which leads to less efficient workarounds, such as multiple sequential or parallel calls to `getUserById()`, increasing complexity and potential API rate limiting issues.

#### Suggested Feature:

Introduce a new method, perhaps named `getUsersByIds()`, which allows passing an array of user IDs to retrieve multiple user records in a single call. The proposed method signature could be:

```javascript
const { data, error } = await supabase.auth.admin.getUsersByIds(['userId1', 'userId2', 'userId3'])
```

Alternatively, enhancing the `supabase.from()` query builder to support filtering by multiple IDs in the `auth.users` table, similar to:

```javascript
const { data, error } = await supabase.from("auth.users").filter('id' , 'in', '("userId1", "userId2")')
```

#### Use Cases:

1. **Batch User Operations**: Efficiently handling batch operations that require user information, reducing the number of API calls.
2. **User Management Dashboards**: Allowing administrators to query multiple user records at once for dashboard displays or bulk updates.
3. **Enhanced User Experience**: Reducing load times and improving responsiveness in applications that need to display multiple user profiles simultaneously.

We believe this enhancement will significantly improve developer productivity and application performance for Supabase users. We look forward to feedback from the Supabase team and the community on this proposal.

**Comment by peterpolman** [comment, marked as answer]
@SamuraiT Doing a Supabase Auth POC and came upon a similar need today. Concluded that an auth trigger should be used to populate a public table with user information for every new user created in the table that's used for authentication purposes. [See docs here](https://supabase.com/docs/guides/auth/managing-user-data?queryGroups=language&language=js#accessing-user-data-via-api)

You will then be able to query this table as you wish. As far as I understand role based access control (RBAC) can be used to override RLS and access multiple rows using your scripts/backend access token.

**Reply by SamuraiT** [reply]
thank you !
